server/api/installer.go

```diff
@@ -82,6 +84,11 @@ func (a *Installer) setRouters() {
 	a.Post("/cancelAllSteps", handler.EnsureAuthenticated(a.WrapHandlerWithDBAndEngine(handler.CancelAllSteps)))
 	a.Post("/deleteContainer", handler.EnsureAuthenticated(a.WrapHandlerWithDB(handler.DeleteContainer)))
 	a.Post("/terminate", handler.EnsureAuthenticated(a.WrapHandlerWithDB(handler.Terminate)))
+	if config.GetEnvironment().AUTH_TYPE == "SSO" {
+		ssoClient := a.loginMgr.(*handler.SsoHandler)
+		a.Get(handler.REDIRECT_PATH, a.WrapHandlerWithDB(ssoClient.SsoRedirect))
```

Comment:
This should go to the UI instead and API added to receive the sso data and return validation/id info.